tests/asynch/basic.rs

```diff
@@ -1,9 +1,103 @@
+use futures::try_join;
 use shuttle::sync::Mutex;
 use shuttle::{asynch, check_dfs, check_random, scheduler::PctScheduler, thread, Runner};
 use std::sync::atomic::{AtomicUsize, Ordering};
 use std::sync::Arc;
 use test_env_log::test;
 
+async fn do_err(e: bool) -> Result<(), ()> {
+    if e {
+        Err(())
+    } else {
+        Ok(())
+    }
+}
+
+// Check that try_join on shuttle futures works as expected
+#[test]
+fn test_try_join() {
+    check_dfs(
+        || {
+            let f2 = do_err(true);
+            let f1 = do_err(false);
+            let res = asynch::block_on(async { try_join!(f1, f2) });
+            assert!(res.is_err());
+        },
+        None,
+    );
+}
+
+// Check that when we block on a future, it gets executed in both executions
+// (block_on task finishing first and main task finishing first).
+#[test]
+fn block_shuttle_future() {
+    let orderings_orig = Arc::new(AtomicUsize::new(0));
+    let async_accesses_orig = Arc::new(AtomicUsize::new(0));
+    let async_accesses = async_accesses_orig.clone();
```

Comment:
nit: make the order of these declarations consistent, and we usually use `foo` and `foo_clone`